Cython/Compiler/Nodes.py

```diff
@@ -1421,6 +1421,8 @@ def analyse_declarations(self, env, dest_scope=None):
                     self.entry.type.create_to_py_utility_code(env)
                     self.entry.create_wrapper = True
             else:
+                if self.overridable:
+                    error(self.pos, "Variables can not be cpdef")
```

Comment:
Sorry, @da-woods. If `error` means raising an error instead of a warning (I'm not familiar with Cython's internal or test system), I have a big 👎. This will break tons of code out there (including ours) that implemented the workaround shown in #3959. I am sure we learned it somewhere so we can't be alone.

Reply:
Your workaround will work just as well with `cdef`. Currently `cpdef` behaves identically to `cdef` I think.

I'd be in favour of sticking with `error` (at least for Cython 3 where some incompatibility is OK) since few people read warnings and the fact that `cpdef` compiles leads people to believe that it does something. But I'll see what other people say....

Reply:
That's a fair point. Perhaps we raise a warning in 0.29.x and turn it into an error in 3.0?

Reply:
I think that's reasonable (although others might have a different view) - I've created https://github.com/cython/cython/pull/3972 to do that for 0.29.x.